### ISSUE TITLE
Post defaults not recognised in Jekyll v3.0.0

### ISSUE THREAD

**Issue by paulrobertlloyd** [jekyll/jekyll#4064]
Given the following provided in config.yml:

```yaml
defaults:
  - scope:
      type: posts
      path: '_posts'
    values:
      layout: entry
      type: entry
      license:
        type: Creative Commons
        url: http://creativecommons.org/licenses/by-nc-sa/4.0/
```

and the following liquid markup:

```html
<pre>Type: {{ page.type }}</pre>
<footer>
{% if page.license %}
    <a rel="license" href="{{ page.license.url }}">{{ page.license.type }}</a>
{% endif %}
</footer>
```

the following is (not) generated:

```html
<pre>Type: </pre>
<footer>
</footer>
```

This was working in versions prior to rc1.

**Comment by parkr** [comment]
What happens if you set `path: ""`?

**Comment by paulrobertlloyd** [comment]
That does work – but of course, my setup requires me to declare different paths:

```yaml
defaults:
  - scope:
      type: content
    values:
      layout: content
      type: content

  - scope:
      type: posts
      path: '_posts/entries'
    values:
      layout: entry
      type: entry
      license:
        type: Creative Commons
        terms: BY-NC-SA
        description: 'You can republish this content without asking permission, but you must credit me (Paul Robert Lloyd) as the author and provide a link back to the original. You may not use any content for commercial purposes. If you make any changes, you must distribute these under the same license as the original.'
        url: http://creativecommons.org/licenses/by-nc-sa/4.0/

  - scope:
      type: posts
      path: '_posts/links'
    values:
      layout: entry
      type: link

  - scope:
      type: posts
      path: '_posts/galleries'
    values:
      layout: gallery
      type: gallery
      theme: dark

  - scope:
      type: projects
    values:
      layout: project
      type: project

  - scope:
      type: talks
    values:
      layout: talk
      type: talk

  - scope:
      type: discussions
    values:
      layout: discussion
      type: discussion
```

**Comment by parkr** [comment]
This is a bit thorny... the defaults _do_ apply. If I access the key from the resulting hash merged in `Document#to_liquid` in Ruby:

<img width="1092" alt="screen shot 2015-10-26 at 5 23 35 pm" src="https://cloud.githubusercontent.com/assets/237985/10746276/5b79e0c2-7c06-11e5-8eac-ead9a78f3250.png">

but when I ask for them via Liquid, I'm denied:

<img width="521" alt="screen shot 2015-10-26 at 5 25 08 pm" src="https://cloud.githubusercontent.com/assets/237985/10746296/9a7a94f6-7c06-11e5-931a-f767c4d8ef93.png">

Looks like Liquid 3 doesn't handle the `default_proc`, perhaps? @fw42, might you know about this?

**Comment by paulrobertlloyd** [comment]
So what’s changed in the released version of Jekyll 3.0 to make this break, as Liquid was happy before? Is this bug related to the posts/collections refactor?

**Comment by parkr** [comment]
/cc @pushrax or @dylanahsmith for my question above:

> Looks like Liquid 3 doesn't handle the default_proc, perhaps?

**Comment by parkr** [comment]
> Where is a default proc given to liquid? I'm not familiar with jekyll

No sweat, sorry I didn't follow that up with a link. The default proc, which is given to the hash which is passed to Liquid, is created here: https://github.com/jekyll/jekyll/blob/cae8bd31c2a9dc976e2c0313739d00f8cca7554a/lib/jekyll/document.rb#L33-L35 Thanks, @dylanahsmith!

**Comment by parkr** [comment]
This is fixed with Drops, I believe. #4277.

**Comment by ccampo133** [comment]
Any idea if this fixes https://github.com/jekyll/jekyll/issues/4076? I'm using Jekyll 3.1.1 and that issue still is occurring for me - that is, custom front matter variables are not available under `page`. For example, if I set `custom_var: foo` in and in default.html I do `{{ page.custom_var }}` it will be `nil`.

**Comment by aviaryan** [comment]
Any estimate when Jekyll 3.1 will be available on github-pages. My site massively relied on custom front-matter variables and it's broken right now. Jekyll v3.0.3 doesn't seem to have this fixed.

**Comment by envygeeks** [comment]
@aviaryan you'll need to ask that to support@github.com or at https://github.com/github/pages, we do not control when they update Jekyll releases even if some people on core are part of the pages team.

**Comment by aviaryan** [comment]
@envygeeks Okay. I emailed them. Btw, in which Jekyll release was the custom front-matter issue fixed ?

**Comment by parkr** [comment]
@ccampo133 Your layouts should have access to custom vars in pages, posts, and collection documents. So if you page has `custom_var: foo`, then your layout should be able to access that via `{{{ page.custom_var }}`

@aviaryan I can see if a fix can be made on the 3.0-stable branch. The review process for pushing out a new version of Jekyll is not easy so it takes a little while. Can you push up an example repository that replicates the issue for you other than the site you mention above?
